src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -222,7 +222,8 @@ internal async Task CreateIfNotExistsAsync(CloudTable table, int retryCount = 3,
                     // though these should only happen in tests not production, because we only ever
                     // delete OLD tables and we'll never be attempting to recreate a table we just
                     // deleted outside of tests.
-                    if (e.RequestInformation.HttpStatusCode == (int)HttpStatusCode.Conflict)
+                    if (e.RequestInformation.HttpStatusCode == (int)HttpStatusCode.Conflict &&
+                        attempt < retryCount)
```

Comment:
It took me a while to catch this (for some reason I *never* hit it while debugging) -- but I was getting 404s while writing a batch later in the flow b/c the table wasn't there. Then I realized that we'd swallow this exception if our retries expired.

Reply:
I like your idea of having the tests pass in via ctor a higher number, and leave the default at 3 for production

Reply:
Okay will do.